### DISCUSSION TITLE
Link spaCy as a static library to be called from C++

### DISCUSSION THREAD

**Discussion by schoedl** [explosion/spaCy#10404] [Help: Coding & Implementations]
Hello,

I just started looking at spaCy. It seems to be mostly written in Cython, with the modules being binary code. Did anyone manage to use spaCy from C/C++ directly, without a Python interpreter? A single binary would make deployment a whole lot easier.

Thanks for any advice,

Arno

**Comment by adrianeboyd** [comment]
It's not exactly what you're asking, but if you'd like to have a single binary, have you looked at building a pex (https://github.com/pantsbuild/pex)? The github repo has a `Makefile` that builds a spacy pex: https://github.com/explosion/spaCy/blob/master/Makefile. You probably want to adjust the settings, e.g., if you want to include additional dependencies like `spacy-transformers`, which isn't included by default.
